test/shared/src/main/scala/zio/test/TestAspect.scala

```diff
@@ -279,6 +285,62 @@ object TestAspect extends TimeoutVariants {
         ZIO.blocking(test)
     }
 
+  /**
+   * An aspect that applies the provided zio aspect to each sample of all checks
+   * in the test.
+   *
+   * i.e.
+   * {{{
+   * test("example") {
+   *   check(Gen.int) { i =>
+   *     ZIO.succeed(assert(i, Assertion.equalTo(1)))
+   *   }
+   * } @@ checks(ZIOAspect.debug)
+   * }}}
+   *
+   * is equivalent to
+   *
+   * {{{
+   * test("example") {
+   *   check(Gen.int) { i =>
+   *     ZIO.succeed(assert(i, Assertion.equalTo(1))) @@ ZIOAspect.debug
+   *   }
+   * }
+   * }}}
+   */
+  def checks(aspect: CheckAspect): TestAspectPoly = checksZIO(
+    ZIO.succeed(aspect)(Trace.empty)
+  )
+
+  /**
+   * An aspect that applies the provided zio aspect to each sample of all checks
+   * in the test. The aspect will be constructed from the provided effect before
+   * each test is run.
+   */
+  def checksZIO[R, E](
+    makeAspect: ZIO[R, E, CheckAspect]
+  ): TestAspect[Nothing, R, E, Any] =
+    new TestAspect[Nothing, R, E, Any] {
+      def some[R1 <: R, E1 >: E](spec: Spec[R1, E1])(implicit trace: Trace): Spec[R1, E1] =
+        spec.transform[R1, E1] {
+          case Spec.TestCase(oldTest, annotations) =>
+            val newTest = makeAspect.mapError(TestFailure.fail).flatMap { aspect =>
+              testConfigWith { oldConfig =>
+                val newConfig = new TestConfig {
```

Comment:
I think we can use the `TestV2` case class here instead. I don't really know why the `Test` case class wasn't used here but I can't think of a good reason